### ISSUE TITLE
Android - 3rd party launchers are unable to run PPSSPP as intent

### ISSUE THREAD

**Issue by arutemu** [hrydgard/ppsspp#17416]
### Game or games this happens in

All games

### What area of the game / PPSSPP

Note - I guess this issue is related to [this one - #17409](https://github.com/hrydgard/ppsspp/issues/17409) or have similar roots.

**Summary:**
[Daijishou](https://github.com/TapiocaFox/Daijishou) - launcher for all kind of emulators - used shortcuts to reach PPSSPP to launch games directly, avoiding main emulator menu.
It worked on PPSSPP 1.14.2
The command used to do that is:
```
-n org.ppsspp.ppsspp/.PpssppActivity
  -a android.intent.action.VIEW
  -d {file.uri}
  --activity-clear-task  --activity-clear-top  --activity-no-history
```
After version PPSSPP 1.14.2 the app is no longer able to launch the emulator and throws en error:
`android.content.ActivityNotFoundException: Unable to find explicit activity class {org.ppsspp.ppsspp/org.ppsspp.ppsspp.PpssppActivity}; have you declared this activity in your AndroidManifest.xml, or does your intent not match its declared <intent-filter>?`

I suppose the issue appeared with changes in **AndroidManifest.xml**.

### What should happen

The external launcher is able to launch a game with PPSSPP.

### Logs

[log from Daijisho.txt](https://github.com/hrydgard/ppsspp/files/11410259/log.from.Daijisho.txt)

### Platform

Android

### Mobile phone model or graphics card

Samsung Galaxy Tab S8, Android 13

### PPSSPP version affected

1.15.2, 1.14.4

### Last working version

1.14.2

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] #17417

**Comment by hrydgard** [comment]
We changed the target Android version, so likely some new restriction came into force. This kind of thing is really annoying, no intentional change has been made to break it.

**Comment by hrydgard** [comment]
There's some advice here that says to use an Action instead of directly specifying the class:

https://www.reddit.com/r/AutomateUser/comments/10eowfo/androidcontentactivitynotfoundexception_unable_to/

I don't know (or remember) enough about this stuff to make sense of it though.

**Comment by arutemu** [comment]
@hrydgard thank you for reply, checking the advice (but with no luck atm 😔).

However, I wanted to remind of the issue #17409 - shortcuts do not work in PPSSPP itself now as well. 
So maybe something really is not working correctly/set up in latest manifest file?

**Comment by arutemu** [comment]
Also, I am writing this without any coding experience, so sorry for stupid questions.
But I don't see this line in the latest manifest file, shouldn't it be included?
`package="org.ppsspp.ppsspp"
`

**Comment by hrydgard** [comment]
The package line should be injected by gradle during build, and Google Play would have rejected the app if that didn't happen..

**Comment by Botosi5150** [comment]
I'm having the same issue with ppsspp
 It gives the unable to find explicit activity class error. I'm on a Samsung galaxy s21 using official apps from the Play Store if any of that helps.

**Comment by unknownbrackets** [comment]
This command worked in a previous version (not long ago), but does not work now:
```sh
am start -n org.ppsspp.ppsspp/.PpssppActivity -a android.intent.action.VIEW -d http://framedump.ppsspp.org/repro/crisis_core_flat_repro.ppdmp
```

But what does "work" (well, the file doesn't load, but...) is:
```sh
am start -n org.ppsspp.ppsspp/.PpssppActivity -a android.intent.action.VIEW -c android.intent.category.DEFAULT -d file:///storage/emulated/0/PSP/ISO/...cso -t application/octet-stream
```

I can make a change to allow more of the previous URLs, which I think we need.  But note that `-c` and `-t` seem to be required now.  I think this is just Android 13.

-[Unknown]

**Comment by arutemu** [comment]
@hrydgard @unknownbrackets after modifying the call command in Daijisho and installing the latest PPSSPP ver available (v1.15.3-4-g90ea4243e) can confirm that the launcher is able to launch games once again. Also - shortucts (widgets) on the homescreen are working again as well. 

Modified command:
```
-n org.ppsspp.ppsspp/.PpssppActivity 
-a android.intent.action.VIEW 
-c android.intent.category.DEFAULT
-d {file.uri} 
-t application/octet-stream
--activity-clear-task  --activity-clear-top  --activity-no-history
```
Thanks for the update!
I just wonder if all required fixing is done and the issue can be closed.

**Comment by hrydgard** [comment]
I don't think there's more to do in PPSSPP, so I'll close. Thanks @unknownbrackets !
